src/rgw/rgw_rest_s3.cc

```diff
@@ -5633,51 +5640,7 @@ AWSGeneralAbstractor::get_auth_data_v4(const req_state* const s,
      *   Version 4 requests. It provides a hash of the request payload. If
      *   there is no payload, you must provide the hash of an empty string. */
     if (!is_v4_payload_streamed(exp_payload_hash)) {
-      ldpp_dout(s, 10) << "delaying v4 auth" << dendl;
-
-      /* payload in a single chunk */
-      switch (s->op_type)
-      {
-        case RGW_OP_CREATE_BUCKET:
-        case RGW_OP_PUT_OBJ:
-        case RGW_OP_PUT_ACLS:
-        case RGW_OP_PUT_CORS:
-        case RGW_OP_PUT_BUCKET_ENCRYPTION:
-        case RGW_OP_GET_BUCKET_ENCRYPTION:
-        case RGW_OP_DELETE_BUCKET_ENCRYPTION:
-        case RGW_OP_INIT_MULTIPART: // in case that Init Multipart uses CHUNK encoding
-        case RGW_OP_COMPLETE_MULTIPART:
-        case RGW_OP_SET_BUCKET_VERSIONING:
-        case RGW_OP_DELETE_MULTI_OBJ:
-        case RGW_OP_ADMIN_SET_METADATA:
-        case RGW_OP_SYNC_DATALOG_NOTIFY:
-        case RGW_OP_SYNC_DATALOG_NOTIFY2:
-        case RGW_OP_SYNC_MDLOG_NOTIFY:
-        case RGW_OP_PERIOD_POST:
-        case RGW_OP_SET_BUCKET_WEBSITE:
-        case RGW_OP_PUT_BUCKET_POLICY:
-        case RGW_OP_PUT_OBJ_TAGGING:
-	case RGW_OP_PUT_BUCKET_TAGGING:
-	case RGW_OP_PUT_BUCKET_REPLICATION:
-        case RGW_OP_PUT_LC:
-        case RGW_OP_SET_REQUEST_PAYMENT:
-        case RGW_OP_PUBSUB_NOTIF_CREATE:
-        case RGW_OP_PUBSUB_NOTIF_DELETE:
-        case RGW_OP_PUBSUB_NOTIF_LIST:
-        case RGW_OP_PUT_BUCKET_OBJ_LOCK:
-        case RGW_OP_PUT_OBJ_RETENTION:
-        case RGW_OP_PUT_OBJ_LEGAL_HOLD:
-        case RGW_STS_GET_SESSION_TOKEN:
-        case RGW_STS_ASSUME_ROLE:
-        case RGW_OP_PUT_BUCKET_PUBLIC_ACCESS_BLOCK:
-        case RGW_OP_GET_BUCKET_PUBLIC_ACCESS_BLOCK:
-        case RGW_OP_DELETE_BUCKET_PUBLIC_ACCESS_BLOCK:
-	case RGW_OP_GET_OBJ://s3select its post-method(payload contain the query) , the request is get-object
-          break;
-        default:
-          ldpp_dout(s, 10) << "ERROR: AWS4 completion for operation: " << s->op_type << ", NOT IMPLEMENTED" << dendl;
-          throw -ERR_NOT_IMPLEMENTED;
```

Comment:
> Other operations not in the switch statement were wrongly rejected as NotImplemented, when they certainly were (e.g. DeleteObject)

my interpretation of this switch statement is that it tracks the ops that call `do_aws4_auth_completion()`. we reject other ops where we haven't implemented this v4 auth completion

there are a bunch of s3 ops which we don't expect to contain a request body, and rgw replies after reading just their headers. maybe we could use a separate switch statement to track those, and perform their 'auth completion' here by comparing their `exp_payload_hash` against the empty hash `e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855`

an alternative is to add explicit calls to `do_aws4_auth_completion()` to more ops, so we can mark them as 'supported' in the existing switch